python/GafferUI/Menu.py

```diff
@@ -636,3 +656,35 @@ def keyPressEvent( self, qEvent ) :
 					return
 
 		QtWidgets.QMenu.keyPressEvent( self, qEvent )
+
+	def __actionHovered( self, action ) :
+
+		# Hovered is called every time the mouse moves
+		if self.__lastHoverAction is not None and action == self.__lastHoverAction() :
+			return
+
+		self.__doLeave()
+
+		self.__lastHoverAction = weakref.ref( action )
```

Comment:
Why a weakref? The menu owns the action already - is there any harm in owning another reference? Weak references are generally for breaking circular references, but I can't see one here. Am I missing something?

Reply:
I was getting 'underlying c++ object already deleted' errors on exit without this.

Reply:
Totally sorted in https://github.com/GafferHQ/gaffer/pull/3360/commits/34f015d55ce9017418e74dc1208ce6077a69a2b2